presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -547,9 +552,13 @@ public long getSplitThreadUsageNanos()
         public ListenableFuture<?> process()
                 throws Exception
         {
+            long splitStartTime = 0;
```

Comment:
I would do `SplitProfile splitProfile = new SplitProfile(System.nanoTime, splitId);` here instead. And move `splitProfiler.add(new SplitProfile(splitStartTime, splitId));` here as well.

In case you believed otherwise, writing code like below in Java guarantees removal of `x` (unlike in some other language):
```
  Object x = ...;
  splitProfiler.add(x);
  try {
  }
  finally {
    splitProfiler.remove(x);
  }
```